pandas/core/indexes/timedeltas.py

```diff
@@ -180,12 +180,15 @@ def _join_i8_wrapper(joinf, **kwargs):
     _datetimelike_methods = ["to_pytimedelta", "total_seconds",
                              "round", "floor", "ceil"]
 
-    __eq__ = _td_index_cmp('__eq__')
-    __ne__ = _td_index_cmp('__ne__', nat_result=True)
-    __lt__ = _td_index_cmp('__lt__')
-    __gt__ = _td_index_cmp('__gt__')
-    __le__ = _td_index_cmp('__le__')
-    __ge__ = _td_index_cmp('__ge__')
+    @classmethod
```

Comment:
shouldn't we define the ``_add_comparison_method`` to take a function and live in ``indexes/base``? to avoid this repeated code?

Reply:
I'm open to this.  It isn't entirely trivial because it references DatetimeIndex, TimedeltaIndex, super... I'll take a look at how much datetimelike-specific logic is there; it may be more at home in indexes.datetimelike.

For now I'd advocate closing this fairly straightforward fix.

Reply:
ok pls add to list (opening an issue is ok too, e.g. if its a bigger thing, not likely to get to anytime soon, etc).

Reply:
Yah there's definitely some logic shared between the timedeltas and datetimes version, but its a little bit deceptive in part because `_to_m8` is defined differently in each module.